code/datums/mind.dm

```diff
@@ -1506,30 +1486,28 @@
 
 /datum/mind/proc/make_Rev()
 	if(SSticker.mode.head_revolutionaries.len>0)
-		// copy targets
+		// copy objective
 		var/datum/mind/valid_head = locate() in SSticker.mode.head_revolutionaries
 		if(valid_head)
 			for(var/datum/objective/mutiny/O in valid_head.objectives)
 				var/datum/objective/mutiny/rev_obj = new
 				rev_obj.owner = src
 				rev_obj.target = O.target
-				rev_obj.explanation_text = "Assassinate [O.target.current.real_name], the [O.target.assigned_role]."
+				rev_obj.explanation_text = O.explanation_text
 				objectives += rev_obj
 			SSticker.mode.greet_revolutionary(src,0)
-	SSticker.mode.head_revolutionaries += src
-	SSticker.mode.update_rev_icons_added(src)
-	special_role = SPECIAL_ROLE_HEAD_REV
 
-	SSticker.mode.forge_revolutionary_objectives(src)
-	SSticker.mode.greet_revolutionary(src,0)
+		SSticker.mode.head_revolutionaries += src
+		SSticker.mode.update_rev_icons_added(src)
+		special_role = SPECIAL_ROLE_HEAD_REV
 
-	var/list/L = current.get_contents()
-	var/obj/item/flash/flash = locate() in L
-	qdel(flash)
-	take_uplink()
-	var/fail = 0
-//	fail |= !ticker.mode.equip_traitor(current, 1)
-	fail |= !SSticker.mode.equip_revolutionary(current)
+	else
+		SSticker.mode.head_revolutionaries += src
+		SSticker.mode.update_rev_icons_added(src)
+		special_role = SPECIAL_ROLE_HEAD_REV
+		SSticker.mode.forge_revolutionary_objectives(src)
+		SSticker.mode.greet_revolutionary(src,0)
+	SSticker.mode.equip_revolutionary(src.current)
```

Comment:
Implied `src`.

Reply:
`SSticker.mode.greet_revolutionary(src,0)`

Is this implied src if the argument is required? or are you looking only at line 1510?

Reply:
Line 1510 only, you do not need `src.`